### PR TITLE
Introduce --includes flag to include necessary classes when generating code coverage

### PR DESCRIPTION
**Pull request by azinneera**
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java [hunk 1]
```diff
@@ -152,6 +152,10 @@ public BuildCommand(Path projectPath, PrintStream outStream, PrintStream errStre
     @CommandLine.Option(names = "--taint-check", description = "perform taint flow analysis")
     private Boolean taintCheck;
 
+    @CommandLine.Option(names = "--include-all", hidden = true,
```
Comment (hevayo) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java:155]: Can we make this useful for general users ? without just having a hidden option

Comment (azinneera) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java:155]: For that, we need to able to get the information in the test report as well. That would need to come up with a design for the report.

File: gradle/javaProject.gradle [hunk 2]
```diff
@@ -201,11 +201,21 @@ check {
...
+
 publishing {
+    def artifactCodeCoverageReport = file("$buildDir/jacoco/test.exec")
     publications {
         mavenJava {
             from components.java
         }
+        if (artifactCodeCoverageReport.exists()) {
+            maven(MavenPublication) {
+                artifact source: artifactCodeCoverageReport, classifier: getCheckedOutGitCommitHash()
```
Comment (manuranga) [inline gradle/javaProject.gradle:216]: It's better to append getCheckedOutGitCommitHash() to `version` or some other parameter and keep `classifier` as `jacoco`. I'll locally test this.

@ThisaruGuruge please note.

**Review by hevayo — APPROVED** [review]

File: gradle/javaProject.gradle (hunk 2 above)
Comment (azinneera) [reply gradle/javaProject.gradle]: Removed the commit hash from the classifier in 1f45bdf

**Review by Ibaqu — APPROVED** [review]

**Review by Dilhasha — APPROVED** [review]

**Pull request merged**